pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -0,0 +1,429 @@
+package clusterdeployment
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/openshift/gcp-project-operator/pkg/gcpclient"
+	hivev1alpha1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubetypes "k8s.io/apimachinery/pkg/types"
+	kubeclientpkg "sigs.k8s.io/controller-runtime/pkg/client"
+	//corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	//metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	//"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	//"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_clusterdeployment")
+
+const (
+	// Operator config
+	operatorNamespace = "gcp-project-operator"
+	controllerName    = "clusterdeployment"
+
+	// clusterDeploymentManagedLabel is the label on the cluster deployment which indicates whether or not a cluster is OSD
+	clusterDeploymentManagedLabel = "api.openshift.com/managed"
+	// clusterPlatformLabel is the label on a cluster deployment which indicates whether or not a cluster is on GCP platform
+	clusterPlatformLabel = "hive.openshift.io/cluster-platform"
+	clusterPlatformGCP   = "gcp"
+	// TODO(Raf) get name of org parent folder and ensure it exists
+	orgParentFolderID = ""
+
+	// secret information
+	gcpSecretName       = "gcp"
+	orgGcpSecretName    = "gcp-project-operator-creds"
+	osServiceAccountKey = "osServiceAccount.json"
+	//
+	osdServiceAccountName = "osdmangedadmin"
+)
+
+var OSDRequiredRoles = []string{
+	"roles/storage.admin",
+	"roles/iam.serviceAccountUser",
+	"roles/iam.serviceAccountKeyAdmin",
+	"roles/iam.serviceAccountAdmin",
+	"roles/iam.securityAdmin",
+	"roles/dns.admin",
+	"roles/compute.admin",
+}
+
+var supportedRegions = map[string]bool{
```

Comment:
It might be better to have shortlist and check `ok:= supportedRegions["region"]`. Easier to maintain